managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1705,7 +1710,8 @@ public void asyncClose(final AsyncCallbacks.CloseCallback callback, final Object
 
         lock.readLock().lock();
         try {
-            if (!individualDeletedMessages.isEmpty()) {
+            if (cursorLedger != null && protobufFormat == ZNodeProtobufFormat.Text
```

Comment:
what if persisting into zk fails? Shouldn't we store `individualDeletedPosition` in both the case `Test` and `Binary`